### PR TITLE
Improve message about OMP_THREAD_NUM on startup

### PR DESCRIPTION
**Pull request by heplesser**
This is a follow-up to #2914. It provides an informational message on startup and a warning on setting the thread/VP number. It also tidies up error handling a little.

### CODE REVIEW

**Comment by heplesser** [comment]
The failing test on macOS will be fixed once #2945 is merged.

File: nestkernel/vp_manager.cpp [hunk 1]
```diff
@@ -143,6 +161,18 @@ nest::VPManager::set_status( const DictionaryDatum& d )
       throw KernelException( msg );
     }
 
+    if ( get_OMP_NUM_THREADS() > 0 and get_OMP_NUM_THREADS() != n_threads )
+    {
+      const std::string tstr = ( n_threads > 1 ) ? "threads" : "thread";
+      std::string msg = "NEST ignores the enviroment variable OMP_NUM_THREADS. The number of threads\n";
+      msg += "in a NEST simulation is controlled only by the kernel attributes /local_num_threads\n";
+      msg += "and /total_num_virtual_procs. ";
```
Comment (nicolossus) [inline nestkernel/vp_manager.cpp:169]:
```suggestion
      msg += "in a NEST simulation is controlled only by the kernel attributes 'local_num_threads'\n";
      msg += "and 'total_num_virtual_procs'. ";
```

**Review by nicolossus — APPROVED** [review]
LGTM! I only have a minor formatting suggestion, see inline comment.

**Review by jougs — APPROVED** [review]
Thanks for this nice clean-up!

**Pull request merged**